---
layout: post
title: Willy's PhD Defense
permalink: /phd_defense/
---

<script type="text/javascript">
function hideStuff(id) {
    // hide the element
    document.getElementById(id).style.display = 'none';
    
}

function showStuff(id) {
    // show the lorem ipsum text
    document.getElementById(id).style.display = 'block';
    
}
 
</script>

  <style>

.tableheader{
  background: #D5D5D5;
  height: 70px;
  font-size: 30px;
}

.tablerow{
  background: #FAFAFA;
  height: 70px;
}

.timedetails{
  width: 180px;
  text-align: center;
}

</style>


<table id="schedule" style="border-collapse: collapse;">
   <thead>
      <tr class="tableheader">
         <th colspan="2">Programme</th>
      </tr>
   </thead>
   <tbody>
      <tr class="tablerow">
         <td class="timedetails">08:30 - 09:00</td>
         <td>COFFEE WELCOME</td>
      </tr>
      <tr class="tablerow">
         <td class="timedetails">09:00 - 09:45</td>
         <td>TALK 1 - <u>Rasmus Heller</u>. <i>Should we do genotyping by sequencing?</i>.  <br><a href="#" onclick="showStuff('abstractRasmus'); return false;">(view abstract)</a>
<span id="abstractRasmus" style="display: none;">
Genotyping by sequencing is becoming increasingly popular as a means of generating population and genomic scale data for non-model organisms. We used a small RADseq data set to examine potential problems or biases in common bioinformatics pipelines, using the site frequency spectrum as an informative summary of the data. We found that a standard RADseq genotyping tool overcalls singletons severely. This can be improved by eschewing genotype calls and using the genotype likelihood directly to infer the SFS. However, even after this the SFS from RADseq data is not identical to one obtained using whole-genome shotgun sequencing on the same individuals. Part of the explanation appears to be that the RADseq protocol produces a non-random subset of the genome which may have a different SFS form the genome-wide one. I discuss some implications of this and ongoing work on the topic.
<a href="#" onclick="hideStuff('abstractRasmus'); return false;">(hide abstract)</a>
</span>
         </td>
      </tr>
    <tr class="tablerow">
       <td class="timedetails">09:50 - 10:35</td>
       <td>TALK 2 - <u>Olivier François</u>. <i>Fast inference of individual ancestry coefficients using geographic data</i>. <a href="#" onclick="showStuff('abstractOFrancois'); return false;">(view abstract)</a>
<span id="abstractOFrancois" style="display: none;">
Geography and landscape are important determinants of genetic variation in natural populations, and several ancestry estimation methods have been proposed to investigate population structure using genetic and geographic data simultaneously. Those approaches are often based on computer-intensive stochastic simulations and do not scale with the dimensions of the data sets generated by high-throughput sequencing technologies. There is a growing demand for faster algorithms able to analyse genomewide patterns of population genetic variation in their geographic context. Here, we present TESS3, a major update of the spatial ancestry estimation program TESS. By combining matrix factorization and spatial statistical methods, TESS3 provides estimates of ancestry coefficients with accuracy comparable to TESS and with run-times much faster than the Bayesian version. In addition, the TESS3 program can be used to perform genome scans for selection, and separate adaptive from nonadaptive genetic variation using ancestral allele frequency differentiation tests. The main features of TESS3 are illustrated using simulated data and analysing genomic data from European lines of the plant species Arabidopsis thaliana.
<a href="#" onclick="hideStuff('abstractOFrancois'); return false;">(hide abstract)</a>
</span>
       </td>
    </tr>
      <tr class="tablerow">
         <td class="timedetails">10:40 - 11:00</td>
         <td> COFFEE BREAK
         </td>
      </tr>
      <tr class="tablerow">
         <td class="timedetails">11:00 - 11:45</td>
         <td>TALK 3 - <u>Mark Beaumont</u>. <i>Expectation Propagation for demographic inference using genome data</i>. <a href="#" onclick="showStuff('abstractMBeaumont'); return false;">(view abstract)</a>
<span id="abstractMBeaumont" style="display: none;">
<a href="#" onclick="hideStuff('abstractMBeaumont'); return false;">(hide abstract)</a>
</span>
</td>
      </tr>    
      <tr class="tablerow">
         <td class="timedetails">11:50 - 13:30</td>
         <td> LUNCH TIME
         </td>
      </tr>
      <tr class="tablerow">
         <td class="timedetails">14:00 - 16:00</td>
         <td>PhD DEFENSE - <u>Willy Rodriguez</u>. <i>Reconstructing the demographic history of populations from genomic data</i>. <a href="#" onclick="showStuff('abstractWilly'); return false;">(view abstract)</a>
<span id="abstractWilly" style="display: none;">
The rapid development of DNA sequencing technologies is expanding the horizons of population genetic studies. It is expected that genomic data will increase our ability to reconstruct the history of populations. While this increase in genetic information will likely help biologists and anthropologists to reconstruct the demographic history of populations, it also poses big challenges. In some cases, simplicity of the model may lead to erroneous conclusions about the population under study. Recent works have shown that DNA patterns expected in individuals coming from structured populations correspond with those of unstructured populations with changes in size through time. As a consequence it is often difficult to determine whether demographic events such as expansions or contractions (bottlenecks) inferred from genetic data are real or due to the fact that populations are structured in nature. Moreover, almost no inferential method allowing to reconstruct past demographic size changes takes into account structure effects. 
In this thesis, some recent results in population genetics are presented: (i) a model choice procedure is proposed to distinguish one simple scenario of population size change from one of structured population, based on the coalescence times of two genes, showing that for these simple cases, it is possible to distinguish both models using genetic information form one single individual; (ii) by using the notion of instantaneous coalescent rate, it is demonstrated that for any scenario of structured population or any other one, regardless how complex it could be, there always exists a panmitic scenario with a precise function of population size changes having exactly the same distribution for the coalescence times of two genes. This not only explains why spurious signals of bottlenecks can be found in structured populations but also predicts the demographic history that actual inference methods are likely to reconstruct when applied to non panmitic populations. Finally, (iii) a method based on a Markov process is developed for inferring past demographic events taking the structure into account. This is method uses the distribution of coalescence times of two genes to detect past demographic changes in structured populations from the DNA of one single individual. Some applications of the model to genomic data are discussed.
<a href="#" onclick="hideStuff('abstractWilly'); return false;">(hide abstract)</a>
</span>
         </td>
      </tr>
      <tr class="tablerow">
         <td class="timedetails">17:00 - 18:00</td>
         <td>POT DE THESE
         </td>
      </tr>
   </tbody>
</table>

<br>
<br>

<table style="border-collapse: collapse;">
   <thead>
      <tr class="tableheader">
         <th colspan="2"> Venue</th>
      </tr>
   </thead>
   <tbody>
   <tr class="tablerow">
         <td colspan="2"> For comming to the INSA, take the subway (line B) and get off at the station "Faculté de Pharmacie". See the map <a href="https://drive.google.com/open?id=1DH6ifyOpMuI8xShMtyhKg2Zh210&usp=sharing">here</a>.
         </td>
      </tr>
   </tbody>
</table>



