<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Applying PSMC to Neandertal data</title>
  <meta name="description" content="Willy Rodriguez github site. Some tutorials related with population genetics.
">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://willyrv.github.io/tutorials/bioinformatics/AltaiNea-psmc.html">
  <link rel="alternate" type="application/rss+xml" title="Willy Rodríguez" href="http://willyrv.github.io/feed.xml" />
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <div class="site-title" ><a class="site-title" href="/">Willy Rodríguez</a> </div>

        <div id="nav">
        <ul>
            <li><a href="/projects" class="">Projects</a></li>
            <li><a href="/tutorials" class="current">Tutorials</a></li>
            <li><a href="/blog" class="">Blog</a></li>
            <li><a href="/cv" class="">CV</a></li>
            <li><a href="/about" class="">About me</a></li>
        </ul>
    </div>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Applying PSMC to Neandertal data</h1>
    <p class="post-meta">Jul 8, 2015</p>
  </header>

  <article class="post-content">
    <h1 id="guidelines-to-apply-the-psmc-to-neandertal-data">Guidelines to apply the psmc to Neandertal data</h1>

<p>In 2014, it was published <em>The complete genome sequence of a Neanderthal from
the Altai Mountains</em> in
<a href="http://www.nature.com/nature/journal/v505/n7481/full/nature12886.html">nature</a>.
As a part of their study, authors did a psmc analysis with the data. I describe
here how I was able to reproduce the analysis at July 3, 2015.
The tools I used where:
* <a href="https://github.com/samtools/samtools">samtools</a> version 1.2-88-gf845404
* <a href="https://github.com/samtools/htslib">htslib</a> version 1.2.1-125-g3a95a67
* <a href="https://github.com/samtools/bcftools">bcftools</a> version 1.2-46-g6baab20</p>

<p><strong>First we download and compile the tools</strong></p>

<p>```
git clone https://github.com/samtools/samtools.git
git clone https://github.com/samtools/htslib.git
git clone https://github.com/samtools/bcftools.git</p>

<p>cd bcftools
make
cd ../samtools
make
```</p>

<p>The read sequences of the AltaiNeandertal where aligned to the the GRCh37 build
of the <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz">human reference</a></p>

<p>Assume we have all the aligned chromosomes in a folder called “bam”, named
<em>chr1.bam</em>, <em>chr2.bam</em> and so on.</p>

<p>We must first uncompress and index the reference file</p>

<p><code>
gunzip ./human_g1k_v37.fasta.gz
./samtools/samtools faidx human_g1k_v37.fasta
</code></p>

<p>This will produce a file <em>human_g1k_v37.fasta.fai</em></p>

<p>Now I run the following pipeline:</p>

<p><code>
./samtools/samtools mpileup -C50 -uf ./human_g1k_v37.fasta ./bam/chr1.bam | \
./bcftools/bcftools call -c | ./bcftools/vcfutils.pl vcf2fq -d 10 -D 100 | \
gzip &gt; ./chr1.fq.gz
</code></p>

<p>Note: Samtools is also able to open a BAM (not SAM) file on a remote FTP or HTTP
 server (see the <a href="http://www.htslib.org/doc/samtools.html">doc</a>). So in the last
 command we could replace <em>./bam/chr1.bam</em> by the link to the bam file in the
 server (i.e. http://cdna.eva.mpg.de/neandertal/altai/AltaiNeandertal/bam/AltaiNea.hg19_1000g.1.dq.bam)</p>

<p>By using the last pipeline I was able to have one fq file per chromosome.
The .fq file produced by vcfutils.pl contains lowercase and uppercase letters (eg
  ‘nnATgagtttAGnn’) wich is a bit confusing given that a fasta sequence (normally)
  should contains only <em>uppercase</em> characters (see the <a href="http://maq.sourceforge.net/fastq.shtml">fastq Format
    Specification</a>). After a bit of
    goggling it seems that lowercase indicates low coverage.</p>

<p>In order to get the input sequence for psmc, I used the <em>fq2psmcfa</em> tool that
comes with the <a href="https://github.com/lh3/psmc">psmc</a> software. For each chromosome</p>

<p><code>
for i in {1..22};
  do ../psmc/utils/fq2psmcfa -q20 ./chr$i.fq.gz &gt; chr$i.psmcfa;
done
</code>
And finally, I put all the chromosomes together in a single file.</p>

<p><code>
for i in {1..22};
  do cat ./chr$i.psmcfa &gt;&gt; AltaiNea.psmcfa;
done
</code></p>

<p>Now I can run the psmc analysis. I use the deffault parameters. Assuming you
have a compiled version of the psmc inside a folder called psmc and your data
(the file AltaiNea.psmcfa) is in the current directory, you can do:</p>

<p><code>
./psmc/psmc -N25 -t15 -r5 -p "4+25*2+4+6" -o ./AltaiNea.psmc ./AltaiNea.psmcfa
</code></p>

<p>This will produce the psmc inferred history (it could take a while). The output
will be written in a file named AltaiNea.psmc</p>

<p>To make a plot and see the inferred demographic history, you can type</p>

<p><code>
./psmc/utils/psmc_plot.pl -p ./AltaiNea.pdf ./AltaiNea.psmc
</code></p>

<p>This will produce a pdf file with a plot of the demographic history.</p>

  </article>

</div>

      </div>
    </div>

  </body>

</html>
